pipeline/tasks/publish_tasks.py

```diff
@@ -56,3 +58,15 @@ def execute(self, repo_url, username, password, publish_env,
 
     def validate(self):
         return []
+
+
+class GitHubPushTask(task_base.TaskBase):
+    """Uploads to a GitHub repository"""
```

Comment:
Let's add details about this task, and emphasize that this task adds and pushes the files in the given directory to a given github repo. It doesn't handle merge and won't delete files (if missing in the local repo) from the remote repo.